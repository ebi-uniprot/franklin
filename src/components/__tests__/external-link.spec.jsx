import { render } from '@testing-library/react';

import ExternalLink from '../external-link';

describe('ExternalLink component', () => {
  test('should render external link', () => {
    const { asFragment } = render(
      <ExternalLink url="#">
        <span>Link text</span>
      </ExternalLink>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render external link without target: _blank attribute', () => {
    const { asFragment } = render(
      <ExternalLink url="#" newTab={false}>
        <span>Link text</span>
      </ExternalLink>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
