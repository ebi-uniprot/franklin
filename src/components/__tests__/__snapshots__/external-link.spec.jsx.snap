// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink component should render external link 1`] = `
<DocumentFragment>
  <a
    class="external-link"
    href="#"
    rel="noopener"
    target="_blank"
  >
    <span>
      Link text
    </span>
    <test-file-stub
      width="12.5"
    />
  </a>
</DocumentFragment>
`;

exports[`ExternalLink component should render external link without target: _blank attribute 1`] = `
<DocumentFragment>
  <a
    class="external-link"
    href="#"
    rel="noopener"
  >
    <span>
      Link text
    </span>
    <test-file-stub
      width="12.5"
    />
  </a>
</DocumentFragment>
`;
